Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -1745,13 +1825,27 @@ infixl:50 " ≃g " => Iso
 
 namespace Hom
 
-variable {G G'} (f : G →g G')
+variable {G G'} {G₁ G₂ : SimpleGraph V} {H : SimpleGraph W} (f : G →g G')
 
 /-- The identity homomorphism from a graph to itself. -/
-abbrev id : G →g G :=
+protected abbrev id : G →g G :=
   RelHom.id _
 #align simple_graph.hom.id SimpleGraph.Hom.id
 
+@[simp, norm_cast] lemma coe_id : ⇑(Hom.id : G →g G) = _root_.id := rfl
+
+instance [Subsingleton (V → W)] : Subsingleton (G →g H) :=
+  FunLike.coe_injective.subsingleton
+
+instance [IsEmpty V] : Unique (G →g H) where
+  default := ⟨isEmptyElim, fun {a} ↦ isEmptyElim a⟩
+  uniq _ := Subsingleton.elim _ _
+
+noncomputable instance [Fintype V] [Fintype W] : Fintype (G →g H) := by
```

Comment:
We shouldn't have noncomputable Fintype instances -- anyone who needs one should introduce a local instance (for example from the `Finite` instance below).

If I'm not mistaken, `[DecidableRel G.Adj]` and `[DecidableRel H.Adj]` suffice to make this computable. You might need `[DecidableEq V]` too if you want to go through `Pi.Fintype` and filter the result.

Reply:
Well technically there are cases where no computable fintype instance can be provided. But I guess this is not the case here.

Reply:
In cases where we can't provide a computable fintype instance, I don't think we should provide a fintype instance at all (at best, we could have a non-instance `def`). If we're willing to have non-computable Fintype instances, we may as well stop worrying about decidability and switch everything to Finite.